dtfs-central-api/src/v1/controllers/tfm/deal-cancellation/tfm-put-update-deal-cancellation.controller.ts

```diff
@@ -0,0 +1,43 @@
+import { Response } from 'express';
+import { flatten } from 'mongo-dot-notation';
+import { HttpStatusCode } from 'axios';
+import { ApiError, ApiErrorResponseBody, AUDIT_USER_TYPES, CustomExpressRequest } from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails, validateAuditDetailsAndUserType } from '@ukef/dtfs2-common/change-stream';
+import { UpdateResult } from 'mongodb';
+import { TfmDealCancellationRepo } from '../../../../repositories/tfm-deals-repo';
+import { PutDealCancellationPayload } from '../../../routes/middleware/payload-validation';
+
+type UpdateTfmDealCancellationRequest = CustomExpressRequest<{
+  reqBody: PutDealCancellationPayload;
+}>;
+
+type UpdateTfmDealCancellationResponse = Response<ApiErrorResponseBody | UpdateResult>;
+
+export const updateTfmDealCancellation = async (req: UpdateTfmDealCancellationRequest, res: UpdateTfmDealCancellationResponse) => {
+  const { dealCancellationUpdate, auditDetails } = req.body;
+  const { dealId } = req.params;
+
+  try {
+    validateAuditDetailsAndUserType(auditDetails, AUDIT_USER_TYPES.TFM);
+
```

Comment:
This happens already when calling `updateOneDealCancellation` - it only matches on deals that don't have `tfm.stage` equal to Cancelled

Reply:
Makes sense @BethThomas141 , but what if this function is called without previously calling `updateOneDealCancellation`

Maybe the check for this (in `updateOneDealCancellation`), should be in this function?

Reply:
So this function will go on to call `updateOneDealCancellation` in the central layer where it will be checked. I think the check there is enough and would prevent us otherwise calling the `getDeals` endpoint from another Repo, finding the matching deal and checking the state here which would add a bit of complexity. 

On the `tfm-ui` layer we won't be showing the deal cancellation button or any of the pages if the deal state is `Cancelled`, so this isn't a scenario we expect unless people are calling endpoints directly, and in either case the check on the central layer will prevent them cancelling the deal.